File: app/src/main/java/com/moudle/app/TranslationDrawableActivity.java
package com.moudle.app;

import android.graphics.drawable.TransitionDrawable;
import android.os.Bundle;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import android.widget.Button;
import android.widget.ImageView;

/**
 * TransitionDrawable   translation 设置图片更换渐变样式
 */
public class TranslationDrawableActivity extends AppCompatActivity {

    private ImageView iv_trans;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_translation);

        iv_trans = findViewById(R.id.iv_trans);
        Button onBtn = findViewById(R.id.btn_on);
        Button offBtn = findViewById(R.id.btn_off);

        onBtn.setOnClickListener(view -> {
            TransitionDrawable trans = (TransitionDrawable) iv_trans.getDrawable();
            trans.startTransition(1500);
        });

        offBtn.setOnClickListener(view -> {
            TransitionDrawable trans = (TransitionDrawable) iv_trans.getDrawable();
            trans.reverseTransition(1500);
        });

    }
}

File: app/src/main/java/com/moudle/app/activity/TranslationDrawableActivity.kt
package com.moudle.app.activity

import android.graphics.drawable.TransitionDrawable
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.moudle.app.databinding.ActivityTranslationBinding

/**
 * TransitionDrawable   translation 设置图片更换渐变样式
 */
class TranslationDrawableActivity : AppCompatActivity() {
    private lateinit var translateBinding:ActivityTranslationBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        translateBinding = ActivityTranslationBinding.inflate(layoutInflater)
        setContentView(translateBinding.root)


        translateBinding.btnOn.setOnClickListener {
            val trans = translateBinding.ivTrans.drawable as TransitionDrawable
            trans.startTransition(1500)
        }

        translateBinding.btnOff.setOnClickListener {
            val trans = translateBinding.ivTrans.drawable as TransitionDrawable
            trans.reverseTransition(1500)
        }
    }
}